code/modules/mob/living/carbon/human/human_movement.dm

```diff
@@ -137,24 +137,28 @@
 	. = ..()
 
 	var/turf/T = loc
-	if (!isturf(T))
-		return
+	var/footsound
+	for(var/obj/structure/S in T)
```

Comment:
```suggestion
	for(var/obj/structure/S in T)
```

Should this maybe check for the structure that is on the highest layer, rather than any structure? I imagine you could get different footstep sounds from one identical turf to the next, if the structures were added in a different order.

Reply:
i definitely could try, but I don't know any procs or whatever that could differentiate the topmost layer.

Reply:
> i definitely could try, but I don't know any procs or whatever that could differentiate the topmost layer.

Can do something like:

```suggestion
	var/top_layer = 0
	for(var/obj/structure/S in T)
		if(S.layer > top_layer && S.footstep_sound)
			top_layer = S.layer
			footsound = S.footstep_sound
```